submit.py

```diff
@@ -339,22 +225,23 @@ def main():
         seen.add(arg)
 
     try:
-        opener = login_from_config(cfg)
+        cfg = get_config()
+        login_reply = login_from_config(cfg)
     except ConfigError as exc:
         print(exc)
         sys.exit(1)
-    except URLError as exc:
-        if hasattr(exc, 'code'):
-            print('Login failed.')
-            if exc.code == 403:
-                print('Incorrect username or password/token (403)')
-            elif exc.code == 404:
-                print("Incorrect login URL (404)")
-            else:
-                print(exc)
+    except requests.exceptions.RequestException as err:
+        print("Login connection failed: " + str(err))
+        sys.exit(1)
+
+    if not login_reply.status_code == 200:
+        print('Login failed.')
+        if login_reply.status_code == 403:
+            print('Incorrect username or password/token (403)')
+        elif login_reply.status_code == 404:
+            print("Incorrect login URL (404)")
         else:
-            print('Failed to connect to Kattis server.')
-            print('Reason: ', exc.reason)
+            print("Status code: {0}".format(login_reply.status_code))
```

Comment:
in the rest of the file we use `print(string1, string2)` and single quotes for string constants.  Would be good to stick to that.  (In addition to using this format you are also using `print(string1 + string2)` in a few places.)